### ISSUE TITLE
[java] UseCollectionIsEmpty can not detect the case this.foo.size()

### ISSUE THREAD

**Issue by wuchiuwong** [pmd/pmd#2543]
**Affects PMD Version:** 
6.22.0
**Rule:**
[UseCollectionIsEmpty](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#usecollectionisempty)

**Description:**
UseCollectionIsEmpty can not detect the case <code>this.foo.size()</code>.
AST analyse result of the code sample:

```xml
<PrimaryPrefix FindBoundary='false' Image='' SingleLine='true' SuperModifier='false' ThisModifier='true' />
<PrimarySuffix ArgumentCount='-1' Arguments='false' ArrayDereference='false' FindBoundary='false' Image='condLink' SingleLine='true' />
<PrimarySuffix ArgumentCount='-1' Arguments='false' ArrayDereference='false' FindBoundary='false' Image='size' SingleLine='true' />
<PrimarySuffix ArgumentCount='0' Arguments='true' ArrayDereference='false' FindBoundary='false' Image='' SingleLine='true'>
	<Arguments ArgumentCount='0' FindBoundary='false' Image='' SingleLine='true' Size='0' />
</PrimarySuffix>
```
In the case <code>if (this.condLink.size() == 0) {</code>, PrimaryPrefix is '', may cause error in comparing whether PrimaryPrefix is collection

**Code Sample demonstrating the issue:**

```java
if (this.condLink.size() == 0) {
```

**Expected outcome:**
false-negative

**Running PMD through:**
CLI
